src/google/protobuf/testing/googletest.cc

```diff
@@ -94,6 +94,10 @@ string TestSourceDir() {
 namespace {
 
 string GetTemporaryDirectoryName() {
+  char *from_environment = getenv("TEST_TMPDIR");
```

Comment:
Please add a comment to explain who is setting this variable (currently bazel only?).

If this is a requirement of bazel, how can we make sure all projects built with bazel are following this?